### PR TITLE
Fix rec pixels not computed when tx is smaller than block

### PR DESCRIPTION
**Pull request by ycho**
The fix applies when tx rdo is turned on.

### CODE REVIEW

**Comment by tdaede** [comment]
Why don't the tests catch this? Can you add a test that fails on master?

**Comment by ycho** [comment]
> Why don't the tests catch this? Can you add a test that fails on master?

I don't think any test can detect this bug? I will point where it has been in next message.

File: src/rdo.rs [hunk 1]
```diff
@@ -1007,7 +1008,7 @@ pub fn rdo_tx_type_decision<T: Pixel>(
         CFLParams::default(), // Unused.
         true,
         rdo_type,
-        true
```
Comment (ycho) [inline src/rdo.rs]: This was the problem, i.e. it has been always true making inverse_tx skipped for some cases.

**Comment by tdaede** [comment]
If the rec pixels fail to be computed, wouldn't that cause a reconstruction / decode mismatch?

**Comment by ycho** [comment]
> If the rec pixels fail to be computed, wouldn't that cause a reconstruction / decode mismatch?

Fortunately, it was during RDO so didn't affect final re-encoding of a block. There is another call of write_tx_blocks() in encode_block_b(), that has correct parameter, i.e. false.

**Comment by tdaede** [comment]
I see, OK. Are there AWCY numbers?

**Comment by ycho** [comment]
> I see, OK. Are there AWCY numbers?

I wanted but due to awcy strike, it is locally running but let me run on awcy right away...

**Comment by ycho** [comment]
AWCY result will soon be available here [AWCY for -s 3](https://beta.arewecompressedyet.com/?job=fix_rec_tx_rdo_s3%402019-05-15T03%3A12%3A38.800Z&job=master_bb8d5690_s3%402019-05-15T03%3A13%3A33.423Z)

**Comment by tdaede** [comment]
It seems to have zero difference... that said, this is probably correct, but I need to think about it a bit.

**Comment by ycho** [comment]
> It seems to have zero difference... that said, this is probably correct, but I need to think about it a bit.

The awcy table showing all zeros might be wrong to me, since the two curves for each metric are broken. I am trying to run locally, this time it should be speed <= 3 since this fix is for tx rdo.

**Comment by ycho** [comment]
From local testing for -s 2,
ducks_take_off_1080p50_60f.y4m
           RATE (%) DSNR (dB)
    PSNR -0.0666773 0.000887102
 PSNRHVS 0.0338921 -0.00202752
    SSIM 0.141132 -0.00325285
FASTSSIM 0.278806 -0.00564569

KristenAndSara_1280x720_60f.y4m
           RATE (%) DSNR (dB)
    PSNR -0.687865 0.0345103
 PSNRHVS -0.489025 0.0309177
    SSIM -0.619933 0.0148676
FASTSSIM -0.300788 0.00302488

AWCY is running for -s 2, so result wll come soon.

**Comment by ycho** [comment]
Hmm, AWCY result for -s2 says there is no change!? Is my local testing wrong?!
[AWCY -s 2](https://beta.arewecompressedyet.com/?job=master_bb8d5690_s2%402019-05-16T17%3A06%3A57.040Z&job=fix_rec_tx_rdo_s2%402019-05-16T17%3A06%3A30.501Z)

**Comment by ycho** [comment]
I forgot the reason why this patch works locally but AWCY shows no change. It needs --tune Psnr/
AWCY uses --tune Psy as default, which makes one of two conditions, i.e. !fi.use_tx_domain_distortion,  for doing inverse tx true.

The fi.use_tx_domain_distortion is set false if --tune Psy is used.
Verifying thru another run on [AWCY for -s 2](https://beta.arewecompressedyet.com/?job=master-0dc0c33_s2_psnr%402019-05-17T22%3A38%3A40.733Z&job=fix_rec_tx_rdo_s2_rebase_psnr%402019-05-17T22%3A39%3A00.151Z)

**Comment by ycho** [comment]
My local testing at -s 2 for KristenAndSara_1280x720_60f.y4m shows:
           RATE (%) DSNR (dB)
    PSNR -0.687865 0.0345103
 PSNRHVS -0.489025 0.0309177
    SSIM -0.619933 0.0148676
FASTSSIM -0.300788 0.00302488

**Comment by ycho** [comment]
|    PSNR | PSNR Cb | PSNR Cr | PSNR HVS |    SSIM | MS SSIM | CIEDE 2000 |
|    ---: |    ---: |    ---: |     ---: |    ---: |    ---: |       ---: |
| -0.4020 | -0.1592 | -0.0924 |  -0.2546 | -0.4039 | -0.3031 |    -0.1819 |

**Comment by ycho** [comment]
I believe this PR is ready to land but let's land it after https://github.com/xiph/rav1e/pull/1284, which renames the existing function parameter name 'for_rdo_use', which has been confusing many of us.

**Comment by lu-zero** [comment]
Yes, please rebase it :)

**Comment by ycho** [comment]
Once testing on awcy for rebased version passes, I will land.

**Comment by ycho** [comment]
[AWCY for -s 1](https://beta.arewecompressedyet.com/?job=test_rec_tx_rdo_s1_psnr_rebase2%402019-05-20T23%3A03%3A59.368Z&job=master-86723c1_s1_psnr%402019-05-20T19%3A35%3A44.653Z)

|    PSNR | PSNR Cb | PSNR Cr | PSNR HVS |    SSIM | MS SSIM | CIEDE 2000 |
|    ---: |    ---: |    ---: |     ---: |    ---: |    ---: |       ---: |
| -0.4352 | -0.3461 |  0.1673 |  -0.2957 | -0.4431 | -0.2883 |    -0.2688 |

[AWCY for -s 0](https://beta.arewecompressedyet.com/?job=test_rec_tx_rdo_s0_psnr_rebase2%402019-05-20T23%3A04%3A09.498Z&job=master-86723c1_s0_psnr%402019-05-20T19%3A35%3A59.907Z)

|    PSNR | PSNR Cb | PSNR Cr | PSNR HVS |    SSIM | MS SSIM | CIEDE 2000 |
|    ---: |    ---: |    ---: |     ---: |    ---: |    ---: |       ---: |
| -0.1063 | -0.6369 | -0.2347 |  -0.0942 | -0.2248 | -0.2032 |    -0.1656 |

**Pull request merged**